pkg/txhandler/txhandler.go

```diff
@@ -0,0 +1,111 @@
+//     - inbound events: these events are received by transaction handler
+//     a. instructional events: transaction handler need to implement the logic to make those event happen on blockchain. (e.g. creating
+//     a new transaction)
+//     b. informational events: transaction handler need to update its own record when an event has already happened to a related
+//     blockchain transaction of its managed transaction
+//     - outbound events: these events are emitted by transaction handler. apitypes.ManagedTransactionEventType contains a list of
+//     managed transaction events that can be emitted by transaction handler.
+//
+// 3. Query functions: functions for querying the current processing status of managed transactions.
+type TransactionHandler interface {
+	// Lifecycle functions
+
+	// Init - setting a set of initialized toolkit plugins in the constructed transaction handler object. Safe checks & initialization
+	//        can take place inside this function as well. It also enables toolkit plugins to be able to embed a reference to its parent
+	//        transaction handler instance.
+	Init(ctx context.Context, toolkit *Toolkit)
+
+	// Start - starting the transaction handler to handle inbound events.
+	// It takes in a context, of which upon cancellation will stop the transaction handler.
+	// It returns a read-only channel. When this channel gets closed, it indicates transaction handler has been stopped gracefully.
+	// It returns an error when failed to start.
+	Start(ctx context.Context) (done <-chan struct{}, err error)
+
+	// Event handling functions
+	// Instructional events:
+	// HandleNewTransaction - handles event of adding new transactions onto blockchain
+	HandleNewTransaction(ctx context.Context, txReq *apitypes.TransactionRequest) (mtx *apitypes.ManagedTX, err error)
+	// HandleNewContractDeployment - handles event of adding new smart contract deployment onto blockchain
+	HandleNewContractDeployment(ctx context.Context, txReq *apitypes.ContractDeployRequest) (mtx *apitypes.ManagedTX, err error)
+	// HandleCancelTransaction - handles event of cancelling a managed transaction
+	HandleCancelTransaction(ctx context.Context, txID string) (mtx *apitypes.ManagedTX, err error)
+
+	// Informational events:
+	// HandleTransactionConfirmed - handles confirmations of blockchain transactions for a managed transaction
+	HandleTransactionConfirmed(ctx context.Context, txID string, confirmations []apitypes.BlockInfo) (err error)
+	// HandleTransactionReceiptReceived - handles receipt of blockchain transactions for a managed transaction
+	HandleTransactionReceiptReceived(ctx context.Context, txID string, receipt *ffcapi.TransactionReceiptResponse) (err error)
+
+	// Query functions
+	// Get a single transaction by ID
+	GetTransactionByID(ctx context.Context, txID string) (transaction *apitypes.ManagedTX, err error)
```

Comment:
I was trying to understand why these moved out of the persistence interface?
Is there a responsibility for the TXHandler here?

Reply:
thinking here is giving txhanlder a control point to apply policy for list transactions. But we can discuss and add this incrementally. I cannot think of a use case